### PR TITLE
IAM AD updates

### PR DESCRIPTION
**Pull request by dantavori**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/34383
relates: https://github.com/demisto/etc/issues/35721

## Description
improved error handling in IAM AD (mainly by clearer error messages).

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 1]
```diff
@@ -782,19 +792,25 @@ def create_user_iam(default_base_dn, args, mapper_out, disabled_users_group_cn):
                                             username=ad_user.get('name'),
                                             details=ad_user,
                                             action=IAMActions.CREATE_USER,
-                                            active=True)
+                                            active=False)  # the user should be activated with the IAMInitADUser script
 
             else:
+                error_code, _ = IAMErrors.BAD_REQUEST
+                error_msg = 'Failed to create user. ' + IAM_ERROR_MESSAGE
```
Comment (anara123) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:799]: You use the same message for different failures, Exceptions. Is it necessarily the same failure reason?

File: Packs/Active_Directory_Query/Scripts/IAMInitADUser/IAMInitADUser.py [hunk 2]
```diff
@@ -1,89 +1,120 @@
...
+        if display_name:
+            email_body += 'Name: ' + display_name + '\n'
 
-        email_body += 'sAMAccountName: ' + sAMAccountName + '\n' \
+        email_body += 'username: ' + username + '\n' \
                       'Email: ' + user_email + '\n' \
                       'Password: ' + password + '\n\n' \
                       'Regards,\nIAM Team'
+    else:
+        subject = f'"IAM - Activate User In Active Directory" incident {inc_id} failed with user {display_name}'
```
Comment (anara123) [inline Packs/Active_Directory_Query/Scripts/IAMInitADUser/IAMInitADUser.py:111]: The email subject & email failure subject should be args with default values to allow the users to modify their message/email

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 3]
```diff
@@ -54,6 +54,12 @@
     "dn", "samaccountname", "cn", "ou"
 ]
 
+IAM_ERROR_MESSAGE = "Please make sure that:\n" \
+                    "* You've added a transformer script which determines the OU where " \
```
Comment (anara123) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:58]: can't we validate before we execute the command that OU is mapped (exists)?

File: Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py [hunk 4]
```diff
@@ -54,6 +54,12 @@
...
+                    "the user will be created, in the Active Directory outgoing mapper, " \
+                    "in the User Profile incident type and schema type, under the \"ou\" field.\n" \
+                    "* You're using LDAPS in the Active Directory (port 636) integration.\n"
```
Comment (anara123) [inline Packs/Active_Directory_Query/Integrations/Active_Directory_Query/Active_Directory_Query.py:61]: Isn't there a specific error when the integration fails on the wrong protocol ?

File: Packs/Active_Directory_Query/Scripts/IAMInitADUser/IAMInitADUser.py (hunk 2 above)
Comment (dantavori) [reply Packs/Active_Directory_Query/Scripts/IAMInitADUser/IAMInitADUser.py]: I added an "email_subject" arg, but for the email body it's more complicated as it contains the account details. FYI, it requires an additional input for the AD User Activation playbook in order to allow configuring the email.

Comment (anara123) [reply Packs/Active_Directory_Query/Scripts/IAMInitADUser/IAMInitADUser.py]: You can pass it with the brackets `{}`, no?

**Review by anara123 — APPROVED** [review]

**Comment by dantavori** [comment]
opened a new PR due to issues with the current branch - https://github.com/demisto/content/pull/12504

**Pull request closed without merging**